pydm/data_plugins/epics_plugins/p4p_plugin_component.py

```diff
@@ -175,8 +177,42 @@ def send_new_value(self, value: Value) -> None:
                     self._timestamp = value.timeStamp.secondsPastEpoch
                     self.timestamp_signal.emit(value.timeStamp.secondsPastEpoch)
 
+    @staticmethod
+    def convert_epics_nttable(epics_struct):
+        result = {}
+        for field in epics_struct.keys():
+            value = epics_struct[field]
+            if isinstance(value, np.ndarray):
+                value = value.tolist()
+            elif isinstance(value, p4p.wrapper.Value):
+                value = Connection.convert_epics_nttable(value)
+            result[field] = value
+        return result
+
+    @staticmethod
+    def set_value_by_keys(table, keys, new_value):
+        if len(keys) == 1:
+            key = keys[0]
+            try: 
+                table[key] = new_value
+            except TypeError:
+                table[int(key)] = new_value
+        else:
+            key = keys[0]
+            if key not in table:
+                raise KeyError 
```

Comment:
This can be removed, the line right below will raise the KeyError for you if it's not valid, as well as what key caused the problem which would be hidden raising it like this without any additional info